pkg/maps/ipcache/ipcache.go

```diff
@@ -173,6 +173,16 @@ func NewMap(name string) *Map {
 	}
 }
 
+// OpenOrCreate wraps the general BPF version to do a probe of LPM support
+// first. Typically NewMap() above is triggered from an init function before
+// the bpf probes map is populated from the BPF probes, so we check it again
+// here. Users should ensure that this OpenOrCreate is invoked after a call
+// to bpf.ReadFeatureProbes().
+func (m *Map) OpenOrCreate() (bool, error) {
```

Comment:
We might consider moving this into the actual `pkg/bpf/` library, there's already fallback code there for handling the LPM->Hash and LRU->Hash map substitutions.